controllers/datadogagent/override/dependencies.go

```diff
@@ -12,20 +12,48 @@ import (
 
 	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
 	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/configmap"
+	"github.com/DataDog/datadog-operator/pkg/kubernetes"
 )
 
 // Dependencies is used to override any resource/dependency settings with a v2alpha1.DatadogAgentComponentOverride.
-func Dependencies(logger logr.Logger, manager feature.ResourceManagers, overrides map[v2alpha1.ComponentName]*v2alpha1.DatadogAgentComponentOverride, namespace string) (errs []error) {
-	for component, override := range overrides {
-		err := overrideComponentDependencies(logger, manager, override, component, namespace)
+func Dependencies(logger logr.Logger, manager feature.ResourceManagers, instance *v2alpha1.DatadogAgent, namespace string) (errs []error) {
```

Comment:
If my previous comment is right, we could keep `overrides map[v2alpha1.ComponentName]*v2alpha1.DatadogAgentComponentOverride` as parameter